clang/lib/CodeGen/CGExprAgg.cpp

```diff
@@ -441,13 +441,13 @@ void AggExprEmitter::EmitWidePointerBitCast(CastExpr *E) {
   RValue SrcRV = CGF.EmitAnyExpr(E->getSubExpr());
   assert(SrcRV.isAggregate());
   Address SrcAddr = SrcRV.getAggregateAddress();
-  EmitWidePointerToDest(E->getType(),
-                        CGF.GetWidePointerElement(SrcAddr, WPIndex::Pointer),
-                        CGF.GetWidePointerElement(SrcAddr, WPIndex::Upper),
-                        E->getType()->isBidiIndexablePointerType()
+  llvm::Value *Ptr = CGF.GetWidePointerElement(SrcAddr, WPIndex::Pointer);
+  llvm::Value *Upper = CGF.GetWidePointerElement(SrcAddr, WPIndex::Upper);
+  llvm::Value *Lower = E->getType()->isBidiIndexablePointerType()
                             ? CGF.GetWidePointerElement(SrcAddr, WPIndex::Lower)
-                            : nullptr,
-                        ElemBitCast);
+                            : nullptr;
+
+  EmitWidePointerToDest(E->getType(), Ptr, Upper, Lower, ElemBitCast);
```

Comment:
@patrykstefanski I suspect this was what caused the Linux code test failures where `getelementptr` instructions were emitted in a different order on Linux.

Reply:
Nice find. After landing this we should try re-enabling the tests we disabled on Linux.